finetune/adapter.py

```diff
@@ -90,7 +90,7 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path):
     config = Config.from_name(name=checkpoint_dir.name)
     checkpoint_path = checkpoint_dir / "lit_model.pth"
     fabric.print(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
-    with fabric.init_module(empty_init=False):
+    with fabric.init_module(empty_init=False), quantization(quantize):
```

Comment:
Can you forbid the combination of quantization plus FSDP? https://twitter.com/Tim_Dettmers/status/1688217887675416577

It should be tested before we advertise support. This applies to all scripts

Reply:
It worked for me, but let me test again after adding the paged optimizer

Reply:
Ah yes, the paged optimizers helps a lot.

From

- 29.20 GB & 762.53 sec 

- to 17.04 GB & 529.83s

for 5k iters using NF4 quantization